test/e2e/main_test.go

```diff
@@ -49,19 +49,38 @@ func TestMain(m *testing.M) {
 		os.Exit(1)
 	}
 
-	// run cleanup before tests, to ensure no resources from previous runs exist.
-	err = framework.CleanUp(defaultNs)
-	if err != nil {
-		log.WithError(err).Error("failed to cleanup resources")
-	}
+	if framework.Namespace == "" {
+		// use a random namespace
```

Comment:
This is not a random function, as the output is easy predictable.
```suggestion
		// use a custom namespace - Unix timestamp
```